### PR TITLE
Removes djangorestframework-jsonapi in favor of a few ember-friendly tweaks

### PR DESCRIPTION
**Pull request by dleehr**
This pull request removes the dependency on djangorestframework-jsonapi.

I attempted to use this to make support for Ember.js easier, but it adds nontrivial overhead to API clients that don't need or strictly conform to JSON-API. We thought that through content negotiation (Accept and Content-type headers), we could expose two interfaces to the same API server, but some required JSON-API conventions (error handling and pagination) can't be disabled this way. Issue #10 was really the tipping point - breaking the login requirement for some query sets.

This pull request:

- Adds a renderer for `Accept: application/vnd.rootobject+json` that wraps JSON responses in a root object (for compatibility with Ember's [DS.RESTAdapter](http://emberjs.com/api/data/classes/DS.RESTAdapter.html#toc_object-root))
- Adds a parser for `Content-type: application/vnd.rootobject+json` that expects incoming JSON payloads to be wrapped in the same root object structure
- Annotates serializers with a `Meta.resource_name` attribute, which is used as the key for the aforementioned root objects.

### CODE REVIEW

**Comment by dleehr** [comment]
Note that when using the web browsable API or an agent accepting `application/json`, the API server will deliver responses without these root objects. This is intentional, to simplify usage for other clients, but if it's simpler to converge on one format, we can do that.

File: data/api.py [hunk 1]
```diff
@@ -36,6 +35,9 @@ def content(self, request, pk=None):
         search_str = request.GET.get('search', '')
         return Response(get_user_project_content(request.user, pk, search_str))
 
+    def get_serializer(self):
+        return self.serializer_class()
+
```
Comment (johnbradley) [inline data/api.py]: Why was this method necessary?

**Review by johnbradley — COMMENTED** [review]
Looks good.

setup_job.sh needs format=json to work after these changes. I will send a commit.

File: data/api.py (hunk 1 above)
Comment (dleehr) [reply data/api.py]: The renderer/serializer use this method to get to the `resource_name` from the serializer. The other views are model-based viewsets that inherit from [generic views](http://www.django-rest-framework.org/api-guide/generic-views/) that implement this method. ProjectsViewSet only inherits from ViewSet, which does not implement `get_serializer()`.

The reason I didn't use generic views here is that they require something for queryset, which I assume only makes sense for model-backed views, but I'll take a look.

File: setup_job.sh [hunk 2]
```diff
@@ -10,34 +10,41 @@ set -e
 TOKEN='FILL_ME_IN'
 # bespin-api url
 BASE_URL=http://127.0.0.1:8000/api
+FORMAT_PARAM=format=json
```
Comment (dleehr) [inline setup_job.sh]: As of cd6bba647a2041d6e6ff99158d31469b4259dde9, the default response should again be JSON. And while requesting the format as a query parameter is supported, it's best to do this with an "Accept: application/json" header.

File: data/api.py (hunk 1 above)
Comment (johnbradley) [reply data/api.py]: Sounds good.

**Comment by johnbradley** [comment]
LGTM.

**Pull request merged**